CleverRaven/Cataclysm-DDA
# Add widgets for vehicle azimuth, cruising speed, and fuel

#### Summary

Features "Add widgets for vehicle azimuth, cruising speed, and fuel"


#### Purpose of change

To show vehicle azimuth (direction), cruising speed (both target and actual), and fuel remaining (as a percentage).


#### Describe the solution

- Add `display::vehicle_azimuth_text` to show the vehicle azimuth (direction) in degrees
- Add `display::vehicle_cruise_text_color` to show the `25 < 20` indicator in color, similar to the classic vehicle speed indicator
- Add `display::vehicle_fuel_percent_text_color` to show color-coded fuel percentage for the currently running engine
- Add `vehicle::engine_fuel_current` to encapsulate getting the current engine's fuel
- Implement widget variables `veh_azimuth_text`, `veh_cruise_text`,`veh_fuel_text` hooked to those `display::` functions

#### Describe alternatives you've considered

There are ways this could be improved upon - having *numerical* widgets for vehicle speed and fuel would enable showing that info more flexibly, and one could imagine other vehicle attributes that might be worth showing in the sidebar.

I thought of adding test cases for this, but there's little precent for tests involving characters actually driving vehicles, and it didn't feel worth the effort just for a widget.


#### Testing

- Configure sidebar with `}` and select the "custom" layout
- Ensure the "Vehicle" section is enabled and visible
- Get in a vehicle and drive around at different speeds - see target/actual speed change
- Get a hose and container; siphon/refill fuel and see fuel percentage change


#### Additional context

Driving around with plenty of gas:

![image](https://user-images.githubusercontent.com/238395/148481726-c4e435f3-a0f4-4e39-9921-0dd811dd4527.png)


Reckless acceleration beyond safe speed:

![image](https://user-images.githubusercontent.com/238395/148481740-4ad5a7c9-1b5a-4b41-b4a1-46f3e27da8f6.png)


Backing up:

![image](https://user-images.githubusercontent.com/238395/148481747-a0e2f690-c69e-465c-918b-abfe3f2fbecb.png)


Low on gas:

![image](https://user-images.githubusercontent.com/238395/148481759-c4d8f442-34d3-4327-b502-2477a5138ace.png)



Split from #53874
